### ISSUE TITLE
Feature Requirement: Snooze for high value only

### ISSUE THREAD

**Issue by marcosdetry** [nightscout/nightguard#39]
Imagine I have the high trigger alarm at 200, so when this threshold is passed by I wake up and give insulin so that the glucose goes back into normal. Usually it would take around 45 to 1 hour to get to normal values, so I have to put the alarm in SNOOSE mode for 1 hour  or half an hour for example.

Sometimes, if I put it in half an hour, it's to little time an I wake up again, although it's going down and in 40 minutes more it would be in range. I would love to be able to put a high SNOOSE time of 1 hour or 1 1/2 hour, but I can't do it because sometimes, over reaction to the correction might put my child in low values and if the Snooze time is longer I might not realise his in a Hypo. 

So basically the feature I request is that if the value that triggered the alarm is bigger than the high value, then when you SNOOSE, the snooze is only for values higher than then HIGH trigger  alarm value, but not for the LOW values.

In this way, If I snooze, for 1 hour an half, and he over reacts and goes below the low trigger, the alarm would go although it's in SNOOSE HIGH MODE.

**Comment by florianpreknya** [comment]
Yes, it can be done, I also find it very useful. I had the same situation two nights before: corrected a 200+ bg, snoozed it for 1hr but I was stressed that a hypo can occur before that hour, so I couldn't sleep at all. It will require a more complex alarm concept, a more complex configuration interface.. so it's a bigger feature. I will talk with @dhermanns about it soon.. I also like the feature that xdrip has: automatocally turn the alarm off if there is a tendency of leaving the high or low zone (for e.g, your bg is 230, the alarm starts; then, the next reading is 219 and the alarm stops because the new bg has a good direction).

**Comment by dhermanns** [comment]
V2.1.0 comes with an improved snooze screen. Not with exactly the requested features (I think they could make the UI quite complex), but better to read and with 45min Snooze.

I think marcos is already a beta tester and therefore should be able to check it out in a few minutes ;-)

**Comment by marcosdetry** [comment]
Yes, I have checked it. Thanks for the 45 min. I think the UI in this version is much better than the prior one. Thanks !

**Comment by marcosdetry** [comment]
I think that the requested feature could be implemented without any change in the Snooze UI, I mean in the UI when you do a SNOOZE.

I would add a toggle in the settings pane between two option. "Total Snooze" or "High Snooze does not invalidate Low Snooze" and may be a info button where a more detailed explanation is given.

I agree that the Snooze function should be as simple as it is today and users should not be chosen what type of snooze they use in the middle of the night. You choose the one you like in settings and then you snooze as normal.

**Comment by dhermanns** [comment]
But in that case, you would always have to keep in mind what was selected
in settings pane. Otherwise you will get unexpected results.

If we add something like that, I would think the user should see the effect
of a snooze directly in the snooze popup.
marcosdetry <notifications@github.com> schrieb am Mo. 30. Apr. 2018 um
11:39:

**Comment by makotila** [comment]
My rough idea for alerts. Current snooze functionality could be as is now, snoozin all alerts for given time. But addition would be that alarms could open a new alarm screen (such as an incoming call or alarm clock) that displays, labeled "Alert", alarm type "High BG", value and threshold eg "13.3 (13) mmol / ml)" or "45 (45) min. ", The function buttons below, for example Snooze (which only affects the type of alarm in question and the OK button to acknowledge the alarm and not to alarm again before the value first falls within the limit).
